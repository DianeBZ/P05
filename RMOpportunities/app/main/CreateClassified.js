import React, { Component } from 'react';
import { Text, Image, View, StyleSheet,Dimensions,TouchableHighlight,BackAndroid,Navigator} from 'react-native';

import AddProductConnection from './AddProductConnection';
import AddDemandConnection from './AddDemandConnection';
import AddOffer from './AddOffer';
import AddDemand from './AddDemand';

import Translation from './Translation';

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;


export default class CreateClassified extends Component {

  render() {
      BackAndroid.addEventListener('Back',this.onBackAndroid);
        return(
        <View>
            <View style = {{height : height*0.08}}/>
            <View style={styles.intro}>
              <Text style={styles.titre}>{Translation[lang].creer_ann}</Text>
            </View>
            <View>
              <Text style={styles.introCommencer}>{Translation[lang].commencer}</Text>
            </View>
            <View style={styles.depotAnnonce}>
              <Text style={styles.presentation}>{Translation[lang].texte_depot} </Text>
            </View>
            <View style={styles.contH}>
              <View style={styles.contV}>
                <Image source={require('../img/ajouterDemande.png')} style={styles.ajouterImage}/>
                <TouchableHighlight onPress={this.onButtonPressDemande} style={styles.button} underlayColor="rgb(138,183,46)">
                    <Text style={styles.buttonText}>{Translation[lang].aj_demande}</Text>
                </TouchableHighlight>
              </View>
              <View style={styles.contV}>
                <Image source={require('../img/ajouterProduit.png')} style={styles.ajouterImage}/>
                <TouchableHighlight onPress={this.onButtonPressProduit} style={styles.button} underlayColor="rgb(138,183,46)">
                    <Text style={styles.buttonText}>{Translation[lang].aj_offre} </Text>
                </TouchableHighlight>
                </View>
            </View>
        </View>
    );
  }
	onButtonPressDemande = () => {
	 BackAndroid.removeEventListener('Exit',this.onBackAndroid);
	const { navigator } = this.props;
	if (navigator) {
        if (connection===0){
            navigator.push({
                name: 'AddDemandConnection',
                component: AddDemandConnection,
            })
        }else if (connection===1){
            navigator.push({
                name: 'AddDemand',
                component: AddDemand,
            })
        }
	}
  };
  onButtonPressProduit = () => {
     BackAndroid.removeEventListener('Exit',this.onBackAndroid);
    const { navigator } = this.props;
    if (navigator) {
        if (connection===0){
            navigator.push({
                name: 'AddProductConnection',
                component: AddProductConnection,
            })
        }else if (connection===1){
            navigator.push({
                name: 'AddOffer',
                component: AddOffer,
            })
        }
    }
  };
  onBackAndroid = () => {
      const { navigator } = this.props;
		if (navigator && navigator.getCurrentRoutes().length > 1) {
			BackAndroid.removeEventListener('Back', this.onBackAndroid);
			navigator.pop();
			return true;
		} else {
			return false;
		}
  };
}

const styles = StyleSheet.create({
  intro:{
    backgroundColor: '#F2F2F2'
  },
  titre:{
       fontSize: 32,
       textAlign: 'center',
       color: '#000000',
       marginTop: 15,
       marginBottom: 10
   },
  introCommencer:{
    fontSize:20,
    textAlign:'center',
    margin: height*0.03,
  },
  depotAnnonce:{
    backgroundColor: '#A4D04A',
  },
  presentation: {
    marginTop: 20,
    marginBottom: 15,
    textAlign: 'center',
    color:"black",
    fontSize: 18,
  },
  contH: {
    alignItems:'center',
    flexDirection : 'row',
    justifyContent: 'center',

  },
  contV: {
    flexDirection : 'column',
    margin:width*0.07,
  },
  ajouterImage:{
    width: 70,
    height: 80,
    margin:width*0.05,
  },
  button:{
      height: height*0.1,
      width: width *0.3,
      justifyContent: 'center',
      backgroundColor:'#A4D04A',
      borderRadius:5,
  },
  buttonText:{
      color: "white",
      textAlign: "center",
      fontSize: 18,
  },
});
