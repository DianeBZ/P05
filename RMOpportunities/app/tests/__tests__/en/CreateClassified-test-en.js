// __tests__/CreateClassified-test-en.js
import 'react-native';
import React from 'react';

global.lang = 'en';
import CreateClassified from '../../../main/CreateClassified';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <CreateClassified />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
