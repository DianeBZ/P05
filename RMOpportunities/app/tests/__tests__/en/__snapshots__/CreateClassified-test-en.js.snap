// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "height": 106.72,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#F2F2F2",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontSize": 32,
          "marginBottom": 10,
          "marginTop": 15,
          "textAlign": "center",
        }
      }
    >
      Start creating a classified
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "margin": 40.019999999999996,
          "textAlign": "center",
        }
      }
    >
      Classified creation
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#A4D04A",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "marginBottom": 15,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
    >
      Once the process of creation is over, our team will verify the information before publishing it online.
       
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "margin": 52.50000000000001,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 80,
            "margin": 37.5,
            "width": 70,
          }
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "backgroundColor": "#A4D04A",
              "borderRadius": 5,
              "height": 133.4,
              "justifyContent": "center",
              "width": 225,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          Add a demand
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "margin": 52.50000000000001,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 80,
            "margin": 37.5,
            "width": 70,
          }
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "backgroundColor": "#A4D04A",
              "borderRadius": 5,
              "height": 133.4,
              "justifyContent": "center",
              "width": 225,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          Add an offer
           
        </Text>
      </View>
    </View>
  </View>
</View>
`;
